velox/vector/SimpleVector.h

```diff
@@ -259,7 +263,7 @@ class SimpleVector : public BaseVector {
     if (rows.isSubset(asciiInfo.asciiSetRows())) {
       return asciiInfo.isAllAscii();
     }
-    ensureIsAsciiCapacity(rows.end());
+    ensureIsAsciiCapacity();
```

Comment:
There was the need to set ensureisAsciiCapacity since we only processed the rows that needed the ascii check. If we are guaranteeing ascii vector size, do we really need to check if we ascii vector has capacity everywhere ?

Reply:
The bug we are fixing stems from the fact that we have a vector of size V with ascii of size A, there A < V.
In this situation a Vector::copy() call can crash and/or trigger ASAN.
So, yes, we want the whole memory backed up when we have ascii. It is a minuscule amount of memory anyway, if that what worries you.

Reply:
No worry is not that. I am just saying that we can guarantee then that ascii size A ( = length_) can be created during vector creation (and on resize) . Then we dont need to check for ensureIsAsciiCapacity since we are always guaranteeing that it will be preallocated with the right capacity.

Reply:
See my answer above for "I am just saying that we resize the selectivity vector only if hasSelections() is true . We dont need to check for that and can resize without that ." question. I assume it is the same one.